src/components/split-button/split-button.component.js

```diff
@@ -0,0 +1,249 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withTheme } from 'styled-components';
+import Icon from '../icon/icon';
+import Button from '../button';
+import StyledSplitButtonContainer, { StyledToggleButton } from './split-button.style';
+import StyledSplitButtonChildrenContainer from './split-button-children.style';
+import { validProps } from '../../utils/ether/ether';
+import OptionsHelper from '../../utils/helpers/options-helper';
+import Events from '../../utils/helpers/events';
+
+export class SplitButton extends React.Component {
+  static propTypes = {
+    /**
+     * Customizes the appearance, can be set to 'primary' or 'secondary'.
+     */
+    as: PropTypes.string,
+
+    /**
+     * A custom value for the data-element attribute
+     */
+    'data-element': PropTypes.string,
+
+    /**
+     * A custom value for the data-element attribute
+     */
+    'data-role': PropTypes.string,
+
+    /**
+     * The additional button to display.
+     */
+    children: PropTypes.node.isRequired,
+
+    /**
+     * Gives the button a disabled state.
+     */
+    disabled: PropTypes.bool,
+
+    // keyNavigation: PropTypes.bool,
+
+    /**
+     * The size of the buttons in the SplitButton.
+     */
+    size: PropTypes.oneOf(OptionsHelper.sizesRestricted),
+
+    /**
+     * The text to be displayed in the SplitButton.
+     */
+    text: PropTypes.string.isRequired,
+
+    /**
+     * The busineess theme passed to the component from the theme provider
+     */
+    theme: PropTypes.object
+  }
+
+  static defaultProps = {
+    as: 'secondary',
+    disabled: false,
+    size: 'medium'
+  }
+
+  static safeProps = ['as', 'disabled', 'size']
+
+  constructor(props) {
+    super(props);
+    this.componentTags = this.componentTags.bind(this);
+    this.additionalButtons = [];
+    this.splitButtons = [];
+    this.listening = false;
+  }
+
+  state = {
+    /**
+     * Tracks whether the additional buttons should be visible.
+     */
+    showAdditionalButtons: false
+  }
+
+  /**
+   * Shows the additional buttons.
+   */
+  showButtons = () => {
+    this.setState({ showAdditionalButtons: true });
+    if (!this.listening) {
+      document.addEventListener('keydown', this.handleKeyDown);
+      this.listening = true;
+    }
+  }
+
+  /**
+   * Hides additional buttons.
+   */
+  hideButtons = () => {
+    this.setState({ showAdditionalButtons: false });
+    if (this.listening) {
+      document.removeEventListener('keydown', this.handleKeyDown);
+      this.listening = false;
+    }
+  }
+
+  activeIndex(node) {
+    return node === document.activeElement;
+  }
+
+  scrollToNextButton(index) {
+    this.additionalButtons[index].focus();
+  }
+
+  handleUpPress(index, length) {
+    const decrementedIndex = index > 0 ? index - 1 : length - 1;
+
+    this.scrollToNextButton(decrementedIndex);
+  }
+
+  handleDownPress(index, length) {
```

Comment:
I reckon you could refactor `handleUpPress` and `handleDownPress` into a single function if you wanted to.

Reply:
Yeah I've ended up removing them all together 🎆